### DISCUSSION TITLE
Docker installation: postgresql invalid password

### DISCUSSION THREAD

**Discussion by toglader** [teslamate-org/teslamate#3159] [Getting started]
I installed teslamate dockers using guide from: https://docs.teslamate.org/docs/installation/docker but it doesn't start. For some reason postgresql somehow fails. Any tips?

`grafana_1    | logger=settings t=2023-03-25T20:57:46.3+0000 lvl=info msg="Starting Grafana" version=8.5.15 commit=be4228db5a branch=HEAD compiled=2022-10-24T18:53:02Z
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="The state of unified alerting is still not defined. The decision will be made during as we run the database migrations"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config loaded from" file=/usr/share/grafana/conf/defaults.ini
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config loaded from" file=/etc/grafana/grafana.ini
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from command line" arg="default.paths.data=/var/lib/grafana"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from command line" arg="default.paths.logs=/var/log/grafana"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from command line" arg="default.paths.plugins=/var/lib/grafana-plugins"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from command line" arg="default.paths.provisioning=/etc/grafana/provisioning"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from command line" arg="default.log.mode=console"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_PATHS_DATA=/var/lib/grafana"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_PATHS_LOGS=/var/log/grafana"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_PATHS_PLUGINS=/var/lib/grafana-plugins"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_PATHS_PROVISIONING=/etc/grafana/provisioning"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_ANALYTICS_REPORTING_ENABLED=FALSE"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_SECURITY_ADMIN_USER=admin"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_SECURITY_ADMIN_PASSWORD=*********"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_SECURITY_DISABLE_GRAVATAR=true"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_SECURITY_ALLOW_EMBEDDING=true"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_USERS_ALLOW_SIGN_UP=false"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_AUTH_ANONYMOUS_ENABLED=false"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_AUTH_BASIC_ENABLED=false"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Config overridden from Environment variable" var="GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS=natel-discrete-panel,pr0ps-trackmap-panel,panodata-map-panel,natel-plotly-panel"
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Path Home" path=/usr/share/grafana
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Path Data" path=/var/lib/grafana
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Path Logs" path=/var/log/grafana
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Path Plugins" path=/var/lib/grafana-plugins
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="Path Provisioning" path=/etc/grafana/provisioning
grafana_1    | logger=settings t=2023-03-25T20:57:46.31+0000 lvl=info msg="App mode production"
grafana_1    | logger=sqlstore t=2023-03-25T20:57:46.31+0000 lvl=info msg="Connecting to DB" dbtype=sqlite3
grafana_1    | logger=migrator t=2023-03-25T20:57:46.38+0000 lvl=info msg="Starting DB migrations"
grafana_1    | logger=migrator t=2023-03-25T20:57:46.4+0000 lvl=info msg="migrations completed" performed=0 skipped=392 duration=574.733µs
grafana_1    | logger=plugin.manager t=2023-03-25T20:57:46.57+0000 lvl=info msg="Plugin registered" pluginId=input
mosquitto_1  | 1679777866: mosquitto version 2.0.15 starting
mosquitto_1  | 1679777866: Config loaded from /mosquitto-no-auth.conf.
mosquitto_1  | 1679777866: Opening ipv4 listen socket on port 1883.
mosquitto_1  | 1679777866: Opening ipv6 listen socket on port 1883.
mosquitto_1  | 1679777866: mosquitto version 2.0.15 running
teslamate_1  | waiting for postgres at database:5432
database_1   | 
database_1   | PostgreSQL Database directory appears to contain a database; Skipping initialization
database_1   | 
database_1   | 2023-03-25 20:57:46.712 UTC [1] LOG:  starting PostgreSQL 14.7 (Debian 14.7-1.pgdg110+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit
database_1   | 2023-03-25 20:57:46.713 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
database_1   | 2023-03-25 20:57:46.714 UTC [1] LOG:  listening on IPv6 address "::", port 5432
grafana_1    | logger=plugin.signature.validator t=2023-03-25T20:57:46.72+0000 lvl=warn msg="Permitting unsigned plugin. This is not recommended" pluginID=panodata-map-panel pluginDir=/var/lib/grafana-plugins/grafana-worldmap-panel-ng
grafana_1    | logger=plugin.manager t=2023-03-25T20:57:46.72+0000 lvl=info msg="Plugin registered" pluginId=panodata-map-panel
grafana_1    | logger=plugin.manager t=2023-03-25T20:57:46.73+0000 lvl=info msg="Plugin registered" pluginId=natel-plotly-panel
grafana_1    | logger=plugin.manager t=2023-03-25T20:57:46.73+0000 lvl=info msg="Plugin registered" pluginId=pr0ps-trackmap-panel
grafana_1    | logger=query_data t=2023-03-25T20:57:46.74+0000 lvl=info msg="Query Service initialization"
database_1   | 2023-03-25 20:57:46.744 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
grafana_1    | logger=live.push_http t=2023-03-25T20:57:46.75+0000 lvl=info msg="Live Push Gateway initialization"
database_1   | 2023-03-25 20:57:46.769 UTC [26] LOG:  database system was shut down at 2023-03-25 20:46:10 UTC
database_1   | 2023-03-25 20:57:46.779 UTC [1] LOG:  database system is ready to accept connections
grafana_1    | logger=http.server t=2023-03-25T20:57:46.91+0000 lvl=info msg="HTTP Server Listen" address=[::]:3000 protocol=http subUrl= socket=
grafana_1    | logger=ngalert t=2023-03-25T20:57:46.91+0000 lvl=info msg="warming cache for startup"
grafana_1    | logger=ngalert.multiorg.alertmanager t=2023-03-25T20:57:46.91+0000 lvl=info msg="starting MultiOrg Alertmanager"
grafana_1    | logger=grafanaStorageLogger t=2023-03-25T20:57:46.95+0000 lvl=info msg="storage starting"
database_1   | 2023-03-25 20:57:48.820 UTC [34] FATAL:  password authentication failed for user "teslamate"
database_1   | 2023-03-25 20:57:48.820 UTC [34] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
database_1   | 2023-03-25 20:57:48.823 UTC [35] FATAL:  password authentication failed for user "teslamate"
database_1   | 2023-03-25 20:57:48.823 UTC [35] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
teslamate_1  | 2023-03-25 20:57:48.839 [error] Postgrex.Protocol (#PID<0.166.0>) failed to connect: ** (Postgrex.Error) FATAL 28P01 (invalid_password) password authentication failed for user "teslamate"
teslamate_1  | 2023-03-25 20:57:48.840 [error] Postgrex.Protocol (#PID<0.165.0>) failed to connect: ** (Postgrex.Error) FATAL 28P01 (invalid_password) password authentication failed for user "teslamate"
database_1   | 2023-03-25 20:57:50.263 UTC [36] FATAL:  password authentication failed for user "teslamate"
database_1   | 2023-03-25 20:57:50.263 UTC [36] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
teslamate_1  | 2023-03-25 20:57:50.267 [error] Postgrex.Protocol (#PID<0.166.0>) failed to connect: ** (Postgrex.Error) FATAL 28P01 (invalid_password) password authentication failed for user "teslamate"
database_1   | 2023-03-25 20:57:51.637 UTC [37] FATAL:  password authentication failed for user "teslamate"
database_1   | 2023-03-25 20:57:51.637 UTC [37] DETAIL:  Connection matched pg_hba.conf line 100: "host all all all scram-sha-256"
teslamate_1  | 2023-03-25 20:57:51.639 [error] Postgrex.Protocol (#PID<0.165.0>) failed to connect: ** (Postgrex.Error) FATAL 28P01 (invalid_password) password authentication failed for user "teslamate"
teslamate_1  | 2023-03-25 20:57:51.762 [error] Could not create schema migrations table. This error usually happens due to the following:
teslamate_1  | 
teslamate_1  |   * The database does not exist
teslamate_1  |   * The "schema_migrations" table, which Ecto uses for managing
teslamate_1  |     migrations, was defined by another library
teslamate_1  |   * There is a deadlock while migrating (such as using concurrent
teslamate_1  |     indexes with a migration_lock)
teslamate_1  | 
teslamate_1  | To fix the first issue, run "mix ecto.create".
teslamate_1  | 
teslamate_1  | To address the second, you can run "mix ecto.drop" followed by
teslamate_1  | "mix ecto.create". Alternatively you may configure Ecto to use
teslamate_1  | another table and/or repository for managing migrations:
teslamate_1  | 
teslamate_1  |     config :teslamate, TeslaMate.Repo,
teslamate_1  |       migration_source: "some_other_table_for_schema_migrations",
teslamate_1  |       migration_repo: AnotherRepoForSchemaMigrations
teslamate_1  | 
teslamate_1  | The full error report is shown below.
teslamate_1  | 
teslamate_1  | ** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 2971ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:
teslamate_1  | 
teslamate_1  |   1. Ensuring your database is available and that you can connect to it
teslamate_1  |   2. Tracking down slow queries and making sure they are running fast enough
teslamate_1  |   3. Increasing the pool_size (although this increases resource consumption)
teslamate_1  |   4. Allowing requests to wait longer by increasing :queue_target and :queue_interval
teslamate_1  | 
teslamate_1  | See DBConnection.start_link/2 for more information
teslamate_1  | 
teslamate_1  |     (ecto_sql 3.9.2) lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
teslamate_1  |     (elixir 1.13.4) lib/enum.ex:1593: Enum."-map/2-lists^map/1-0-"/2
teslamate_1  |     (ecto_sql 3.9.2) lib/ecto/adapters/sql.ex:1005: Ecto.Adapters.SQL.execute_ddl/4
teslamate_1  |     (ecto_sql 3.9.2) lib/ecto/migrator.ex:677: Ecto.Migrator.verbose_schema_migration/3
teslamate_1  |     (ecto_sql 3.9.2) lib/ecto/migrator.ex:491: Ecto.Migrator.lock_for_migrations/4
teslamate_1  |     (ecto_sql 3.9.2) lib/ecto/migrator.ex:403: Ecto.Migrator.run/4
teslamate_1  |     (ecto_sql 3.9.2) lib/ecto/migrator.ex:146: Ecto.Migrator.with_repo/3
teslamate_1  |     (teslamate 1.27.2) lib/teslamate/release.ex:9: anonymous fn/2 in TeslaMate.Release.migrate/0`

**Comment by SiNaPsEr0x** [comment]
some problem
